docs/08_jenkins_testing_enviroment.md

```diff
@@ -0,0 +1,130 @@
+
+> What is the project blocked on?
+
+No blockers.
+
+> What will be impacted by the project?
+
+Testing of the Jenkins integration will be impacted.
+
+# Testing
+
+> How will the project be tested?
+
+This project will be tested manually.
+
+# Alternatives Considered
+
+> Summarize alternative designs (pros & cons)
+
+* Setups Jenkins server on Amazon AWS.
+    - Pros:
+         - The AWS instance is free for one year.
+         - Has an excellent [setup guide](https://d1.awsstatic.com/Projects/P5505030/aws-project_Jenkins-build-server.pdf) that explains how to create an instance and deploy the Jenkins server.
+         - Easy to maintain. You could just connect to the AWS instance and use it and use it the way that you would use a computer in front of you.
+    - Cons:
+         - You need to configure the AWS instance and firewall rules manually via the user interface (no automation available), which takes a bit more time in comparison to using the Google Kubernetes deployment.
+         
+         
+* Setup Jenkins on Google Kubernetes Engine.
+    - Pros:  
+         - Easy to set up. The configuration process is fully automated, and it is needed just to configure Jenkins projects.
+           It is needed just to go through the [Jenkins Google cloud setup flow](https://console.cloud.google.com/marketplace/details/google/jenkins).
+         - Jenkins has a [free application](https://console.cloud.google.com/marketplace/details/google/jenkins) on the Google Cloud.
+    - Cons: 
+         - The Kubernetes Engine is the paid feature of the Google Cloud that uses a couple of other paid features like Compute Engine.
+         - Hard to maintain because you cannot just SSH connect to the instance with the Jenkins server to make any changes. For connection, you need to use the "kubectl" CLI.
+         
+         
+* Serve the Jenkins server using the local machine and ngrok.
+    - Pros:
+         - Easy to setup. You need just to run the Jenkins locally and set up the tunnel to the localhost using the ngrok for other users to be able to connect.
+    - Cons: 
+         - Because Jenkins stores the data (pipeline configurations, build, build artifacts, etc.) locally,
+          using the file system, the Jenkins instance should be running on the same machine all the time.
+         - The server will be very dependent on that configured server availability (app availability, computer availability).
+         
+         
+* Setup Jenkins on Google Compute Engine.
+    - Pros:
+         - Easy to connect to the container using SSH to make any changes to the Jenkins.
+    - Cons: 
+         - Pretty complicated setup mechanism because we have a lot of configuration items.
+         - We should set up the Jenkins storage manually in some other place like [Cloud Storage](https://cloud.google.com/storage)
+          because the Compute Engine instance has too few memory to store all tools, required by Jenkins, and builds and artifacts on the instance.
```

Comment:
```suggestion
          because the Compute Engine instance has too few memory to store all tools, required by Jenkins, and builds with artifacts on the instance.
```